torch_xla/_internal/rendezvous.py

```diff
@@ -28,7 +28,7 @@ def pjrt_rendezvous_handler(url: str,
     ) == 'TPU' else 'localhost'
 
   master_port = xu.getenv_as('MASTER_PORT', int, 12355)
-  world_size = xr.world_size()
+  world_size = xr.process_count()
```

Comment:
This should stay `xr.world_size` for the multiprocessing case. This is only returned at the very end to set the `torch.distributed` world size correctly for multithreading, so you can make this conditional and move it down with `rank`

Reply:
Ah I forgot about the multithreading case... Sounds good, I'll ensure this method is exactly the same without SPMD. Thanks for checking Will!